app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt

```diff
@@ -174,18 +205,25 @@ fun NavigationController(
             val cid = backStackEntry.arguments?.getString("challengeId")!!
             ClaimChallenge(
                 cid = cid,
-                onFailure = {
-                    Toast.makeText(context, "Something went wrong, failed to create challenge", Toast.LENGTH_LONG).show()
-                    Log.e("GeoHunt", "Fail to create challenge: $it")
-                    navController.popBackStack()
-                },
+                onFailure = onFailure,
                 onClaimSubmitted = {
                     navController.popBackStack()
                     navController.navigate("challenge-view/$cid")
                 }
             )
         }
 
+        // Bounty
+        composable(
+            "bounty-admin-page/{bountyId}",
+            arguments = listOf(navArgument("bountyId") { type = NavType.StringType })
+        ) { backStackEntry ->
+            val bid = backStackEntry.arguments?.getString("bountyId")!!
+            AdminBountyPage(bid, onFailure = onFailure, onCreateChallenge = {
+                navController.navigate("create-challenge-bounty/$bid")
```

Comment:
Use a HiddenRoute instead of just hard-coding it here

Reply:
Same as above